src/courtroom.cpp

```diff
@@ -3677,76 +3677,42 @@ void Courtroom::handle_song(QStringList *p_contents)
   if (f_contents.size() < 2)
     return;
 
+  bool looping = false; // No loop due to outdated server using serverside looping
+  int channel = 0; // Channel 0 is 'master music', other for ambient
+  int effect_flags = 0; // No effects by default - vanilla functionality
+
   QString f_song = f_contents.at(0);
   QString f_song_clear = f_song.left(f_song.lastIndexOf("."));
   if (f_song.startsWith("http")) {
     QByteArray f_song_bytearray = f_song.toUtf8();
     QString f_song_decoded = QUrl::fromPercentEncoding(f_song_bytearray);
     f_song_clear = f_song_decoded.left(f_song_decoded.lastIndexOf("."));
   }
-  f_song_clear = f_song_clear.right(f_song_clear.length() -
-                                    (f_song_clear.lastIndexOf("/") + 1));
+  f_song_clear = f_song_clear.right(f_song_clear.length() - (f_song_clear.lastIndexOf("/") + 1));
+
   int n_char = f_contents.at(1).toInt();
 
-  // Assume the song doesn't loop unless told otherwise (due to most outdated
-  // servers handling looping through serverside)
-  bool looping = false;
-  // Channel 0 is the 'master music', other channels would commonly be used for
-  // ambience
-  int channel = 0;
-  // No effects assumed by default - vanilla functionality
-  int effect_flags = 0;
-
-  if (n_char < 0 || n_char >= char_list.size()) {
-    int channel = 0;
-    if (p_contents->length() > 3 && p_contents->at(3) == "1")
-      looping = true;
-
-    if (p_contents->length() >
-        4) // eyyy we want to change this song's CHANNEL huh
-      channel = p_contents->at(4).toInt(); // let the music player handle it if
-                                           // it's bigger than the channel list
-
-    if (p_contents->length() > 5) // Flags provided to us by server such as Fade
+  if (p_contents->length() > 3 && p_contents->at(3) == "1")
+    looping = true;
+
+  if (p_contents->length() > 4) // eyyy we want to change this song's CHANNEL huh
+    channel = p_contents->at(4).toInt(); // let the music player handle it if
+                                         // it's bigger than the channel list
+
+  if (p_contents->length() > 5) { // Flags provided to us by server such as Fade
                                   // In, Fade Out, Sync Pos etc.
-    {
-      effect_flags = p_contents->at(5).toInt();
-    }
-    music_player->play(f_song, channel, looping, effect_flags);
-    if (f_song == "~stop.mp3")
-      ui_music_name->setText(tr("None"));
-    else if (channel == 0) {
-      if (file_exists(ao_app->get_sfx_suffix(ao_app->get_music_path(f_song))) && !f_song.startsWith("http"))
-        ui_music_name->setText(f_song_clear);
-      else if (f_song.startsWith("http"))
-        ui_music_name->setText(tr("[STREAM] %1").arg(f_song_clear));
-      else
-        ui_music_name->setText(tr("[MISSING] %1").arg(f_song_clear));
-    }
+    effect_flags = p_contents->at(5).toInt();
   }
-  else {
+
+  bool is_stop = (f_song == "~stop.mp3");
```

Comment:
this is a nitpick but maybe define `is_stop` at the top with the rest of the variables

Reply:
Can't do as f_song is declared after the defaults are set. I would have to nest it in the middle of reading data, which reduces visibility.